### DISCUSSION TITLE
Child Prefix showing available when parent is utilised?

### DISCUSSION THREAD

**Discussion by CelticWebSolutions** [netbox-community/netbox#16262] [Q&A]
Hi all,

I've recently started experimenting with Netbox, iPam is one of the thinsg I think would be very useful but I can't quote get my head around how it functions.

The way I have set it up so far is that top level IP Allocations, such as a range registered at Ripe are set as Aggregates. Then everything below is set as prefixes, with ranges set when a part prefix is used and single ips set under in IP addresses.  Initially this seemed the sensible way for it to work.

Here's where it starts to confuse me a little, I split up a prefix in to all the child prefixes, /28 with /29 below, /30 below that.  The way I looked at this was that if I was, sometimes a system may use a full /28, while other times it may be allocated differently for a use case and maybe only the  /30s out of it for switches.  

I assumed that separating it up in all the child prefixes, I'd immediately be able to see which ones were available and which ones were already used via other allocations. so the /30 in a set group would show as allocated because it's parent was fully utilised.  I think I'm allocating things incorrectly.  Even if I set the /28 as allocated, which shows as fully utilised, the /29 and /30 below it show as available?  So I tried setting a range as utilised instead, again this then shows the /28 as utilised but didn't mark the child /29 and /30 as utilised.

So my question is hopefully a simple one, obviously if a /28 is fully utilised, the /29 and /30 below it would also be fully utilised and unavailable.  Am I missing a setting somewhere that automatically marks child prefixes as used if a parent prefix is allocated and utilised?  Or am I doing this totally wrong and there is a better was of being able to show usage?

Thanks in advance for your help here, I just can't get my head around it without thinking that as I chop and change IP's I have to keep adding and removing prefixes.

**Comment by Mailstorm-ctrl** [comment]
What exactly is your goal? This doesn't sound like a standard setup. You should really have the prefixes you are actually using defined. And then assign IPs within that prefix appropriately. 

Anyways, what you're experiencing is expected.
https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-does-netbox-require-every-ip-to-have-a-mask

Basically, when you fill the top prefix, those IPs are using the mask of that prefix. Netbox is not going to assume that because the "top level" prefix is filled the others below are.

**Reply by CelticWebSolutions** [reply]
Hi,

Probably an odd way of doing it, but when allocating subnets, if I'm asked for a /30 for example, it would be nice to see all the /30 that are currently available then make a choice as to which is most suitable with the least IP wastage. 

IPAM was originally set with prefixes showing

25 - shown as aggregate with the following prefixes as part of it.

/26/26
/27/27/27/27/
28/28/28/28/28/28/28/28
/29/29/29/29/29/29/29/29/29/29/29/29/29/29/29/29
/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30/30

Sometimes a customer will want just a /30. While others may want a /28 for example.  The idea was that we could see exactly what we had available, then allocate appropriately. Then if a /28 is allocated to a customer, it is marked as fully utilised.  What I expected was that when a parent was marked as utilised, all the child prefixes below it would be auto marked as utilised too.  It does not do this.  It shows child prexies of fully utilised parents as being available to be allocated. So the only way to do this is to delete all the child prefixes when a parent prefix is allocated, Then when the parent prefix is relinquished and somebody wants a smaller part of that prefix, add child prefixes back in again.

I think I'm just using it incorrectly because to me that made perfect sense, but it obviously doesn't.

**Reply by Mailstorm-ctrl** [reply]
I'm not a huge networking nerd but it sounds like to me that you should be using VRFs. Doing it the way you are would mean you quickly run out of IP space not to mention other issues that would arise from this.
